### PR TITLE
MSEARCH-620 Another tenant name is displayed in "Held by" facet when …

### PR DESCRIPTION
**Pull request by andrei-razumnov**
Another tenant name is displayed in "Held by" facet when location from current tenant is selected

### Purpose
_Keep filter that we have in request if `term`s are `"items.effectiveLocationId"` `"holdings.tenantId"`_

### Approach
_Add new method `cleanUpFacetSearchSource ` which has list of terms to keep._

### Changes Checklist
- [ ] **API Changes**: List any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Specify any database schema changes and their impact.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Note added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [ ] **NEWS**: Ensure that the `NEWS` file is updated with relevant information about the changes made in this pull request.

### Related Issues
_[MSEARCH-620](https://issues.folio.org/browse/MSEARCH-620)_

### CODE REVIEW

File: src/main/java/org/folio/search/service/FacetService.java [hunk 1]
```diff
@@ -37,17 +39,26 @@ public FacetResult getFacets(CqlFacetRequest request) {
...
+    if (query instanceof BoolQueryBuilder boolQuery) {
+      List<QueryBuilder> filtersToKeep = Collections.emptyList();
+      if (CollectionUtils.isNotEmpty(filterNamesToKeep)) {
+        filtersToKeep = boolQuery.filter().stream()
+          .filter(TermQueryBuilder.class::isInstance)
+          .filter(filter -> filterNamesToKeep.contains(((TermQueryBuilder) filter).fieldName()))
+          .toList();
+      }
+      boolQuery.filter().clear();
+      boolQuery.filter().addAll(filtersToKeep);
```
Comment (viacheslavkol) [inline src/main/java/org/folio/search/service/FacetService.java:59]:
```suggestion
      boolQuery.filter().removeIf(filter -> !(filter instanceof TermQueryBuilder termFilter && filterNamesToKeep.contains(termFilter.fieldName())));
```

File: src/main/java/org/folio/search/service/FacetService.java [hunk 2]
```diff
@@ -37,17 +39,26 @@ public FacetResult getFacets(CqlFacetRequest request) {
     searchSource.size(0).from(0).fetchSource(false);
 
     facetQueryBuilder.getFacetAggregations(request, searchSource.query()).forEach(searchSource::aggregation);
-    cleanUpFacetSearchSource(searchSource);
+    cleanUpFacetSearchSource(searchSource, List.of("items.effectiveLocationId"));
```
Comment (viacheslavkol) [inline src/main/java/org/folio/search/service/FacetService.java:42]: Would be good to move to a constant.
It was mentioned to keep also tenantId filter.

File: src/test/java/org/folio/search/service/FacetServiceTest.java [hunk 3]
```diff
@@ -66,6 +67,28 @@ void getFacets_positive() {
     assertThat(actual).isEqualTo(new FacetResult());
   }
 
+  @Test
+  void getFacets_positive_keepGivenFiler() {
```
Comment (viacheslavkol) [inline src/test/java/org/folio/search/service/FacetServiceTest.java:71]: Typo

**Comment by psmagin** [comment]
Please follow the commit message format described in https://wiki.folio.org/display/FOLIJET/Conventional+Commit+Initiative when you will squash and merge your PR.

**Comment by psmagin** [comment]
Please update NEWS file

**Review by viacheslavkol — APPROVED** [review]

**Review by psmagin — APPROVED** [review]

**Pull request merged**